src/client/windshaft.js

```diff
@@ -338,6 +350,9 @@ export default class Windshaft {
         }
         const response = await fetch(endpoint(conf), this._getRequestConfig(mapConfigAgg));
         const layergroup = await response.json();
+        if (!response.ok) {
+            throw new Error(`Maps API error: ${JSON.stringify(layergroup)}`);
```

Comment:
I'm always a little confused when promises are involved... so in this case it's preferable to cause the exception here than returning a rejected promise?

Reply:
Yes, here is safe because we are inside an `async` function, therefore, thrown exceptions will automatically reject the promise that the function returns.